### ISSUE TITLE
Crossdock test client fails to get local ip

### ISSUE THREAD

**Issue by breerly** [yarpc/yarpc-node#47]
Experienced in #45 

Node is getting undefined for local pub ip, which results in Crossdock not being able to start.

```
yarpc-node2_1 | MY LOCAL IP undefined
yarpc-node2_1 | 
yarpc-node2_1 | assert.js:89
yarpc-node2_1 |   throw new assert.AssertionError({
yarpc-node2_1 |   ^
yarpc-node2_1 | AssertionError: Expected host to be a string
yarpc-node2_1 |     at TChannel.listen (/src/node_modules/tchannel/channel.js:632:9)
yarpc-node2_1 |     at TChannelInbound.start (/src/transport/tchannel/inbound.js:29:18)
yarpc-node2_1 |     at RPC.start (/src/rpc.js:173:26)
yarpc-node2_1 |     at App.start (/src/test/subject.js:218:14)
yarpc-node2_1 |     at TestHarness.start (/src/test/harness.js:61:18)
yarpc-node2_1 |     at main (/src/test/harness.js:147:13)
yarpc-node2_1 |     at Object.<anonymous> (/src/test/harness.js:164:2)
yarpc-node2_1 |     at Module._compile (module.js:409:26)
yarpc-node2_1 |     at Object.Module._extensions..js (module.js:416:10)
yarpc-node2_1 |     at Module.load (module.js:343:32)
yarpc-node_1  | MY LOCAL IP 172.18.0.3
yarpc-node_1  | TCHANNEL 172.18.0.3 8082
yarpc-node_1  | {"log":"YARPC listening for requests","level":"info","transport":"http","listeningAddress":"::","listeningPort":8081}
yarpc-node_1  | {"log":"YARPC listening for requests","level":"info","transport":"tchannel","hostPort":"172.18.0.3:8082","channelListened":true,"channelListening":true,"channelDestroyed":false,"channelDraining":false}
yarpc-node_1  | {"log":"test harness listening for requests","level":"info","address":"0.0.0.0","family":"IPv4","port":8080}
crossdock_1   | 
crossdock_1   | Crossdock starting...
crossdock_1   | 
crossdock_1   | Waiting on CROSSDOCK_CLIENTS=[yarpc-node yarpc-node2]
crossdock_1   | 
yarpcnode_yarpc-node2_1 exited with code 1
crossdock_1   | HTTP: Service yarpc-node:8080 is up
crossdock_1   | 2016/03/15 17:40:57 HTTP: Service yarpc-node2:8080 timed out. Last error: Head http://yarpc-node2:8080/: dial tcp: lookup yarpc-node2 on 127.0.0.11:53: no such host
crossdock_1   | 2016/03/15 17:40:57 Error: One or more services timed out after 30000000000 second(s)
yarpcnode_crossdock_1 exited with code 1
```

**Comment by breerly** [comment]
The fix is to poll network interfaces until local pub ip is not `default`.

Unfortunately, polling the `my-local-ip` exposed func doesn't work because it caches the network interfaces at module import-time: https://github.com/dominictarr/my-local-ip/blob/master/index.js#L3

```node
#! /usr/bin/env node

var n = require('os').networkInterfaces()

var myIp = module.exports = function () {
  var ip = []
  for(var k in n) {
    var inter = n[k]
    for(var j in inter)
      if(inter[j].family === 'IPv4' && !inter[j].internal)
        return inter[j].address
  }
}

if(!module.parent)
  return console.log(myIp())
```

**Comment by breerly** [comment]
Fixed in #45 by polling the network interfaces
